add criterion for avoiding shared xpubs

kristovatlas opened the pull request:
For issue
https://github.com/OpenBitcoinPrivacyProject/wallet-ratings/issues/27

Added criterion CR150
(013d799c49176b6aa9e281058f578c52382bc0d1867850c693e168ea13e37dfb) and
listed it under CM41
(7095f1333780e51e8d15681bd64f3699c78839e1f9d8dcbb4593240d7d2388b3).

Commit:
add criterion for avoiding shared xpubs

For issue
https://github.com/OpenBitcoinPrivacyProject/wallet-ratings/issues/27

Added criterion CR150
(013d799c49176b6aa9e281058f578c52382bc0d1867850c693e168ea13e37dfb) and
listed it under CM41
(7095f1333780e51e8d15681bd64f3699c78839e1f9d8dcbb4593240d7d2388b3).

dcousens commented:
> The client does not use a protocol that shares the same non-ECDH public key with more than one transaction participant

The client does not use a protocol for sharing EC public keys that have a predictable derivation schema.

dcousens commented:
If a client uses an ECDH scheme for sharing public keys,  they must use a shared secret that cannot be used to reveal other participants.

dcousens commented:
The client does not share the same BIP32 extended public key [for the purposes of key derivation] between multiple transaction participants.

kristovatlas commented:
The client uses a secure ECDH protocol to derive payment addresses that are unpredictable to parties outside of the transaction participant.

kristovatlas commented:
The client derives a unique payment address for each payment using random data or a secret root public and a hardened derivation.

kristovatlas commented:
Latest changes are in line with what we agreed on during tonight's meeting.

Commit:
re-word CR150, add CR151 and CR152

covers these 3 comments on the existing PR:
https://github.com/OpenBitcoinPrivacyProject/threat-model-scoring-system
/pull/28#issuecomment-247196375
https://github.com/OpenBitcoinPrivacyProject/threat-model-scoring-system
/pull/28#issuecomment-247197416
https://github.com/OpenBitcoinPrivacyProject/threat-model-scoring-system
/pull/28#issuecomment-247198903
